### PR TITLE
Frontend Logic Rewrite

### PR DESCRIPTION
**Pull request by jooohhn**
**Goal**: Simplify the currently convoluted front-end logic. Will make implementing table schema manipulation easier.

### Checklist
- [x] Make `HomePage` handle only one database file
  - Currently expects an array of databases
- [ ] ~~Move `StructureView` out of `TableView`~~
  -  It's ok if `TableView` wraps `StructureView`.  Done so that controls can be used to switch with both views. Can reconsider later.
- [ ] ~~Move `databaseApi` out of TableView and into `HomePage`. Reason is so that `StructureView` and `TableView` can share a databaseApi, thus app won't have to create two connections.~~
   - Don't need to anymore since StructureView can stay in TableView. 
   - Logic for pooling connections shouldn't be here. See #21 
- [x] Fix flow type issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
   "scripts": {
     "test": "cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 node --trace-warnings ./test/runTests.js",
     "test-all": "npm run lint && npm run flow && npm run test && npm run build && npm run test-e2e",
-    "test-watch": "npm test -- --watch",
+    "test-watch": "npm test --watch",
     "test-e2e": "rm -f test/e2e/temp.sqlite && cp -fR test/e2e/demo.sqlite test/e2e/temp.sqlite && cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 node --trace-warnings ./test/runTests.js e2e",
     "lint": "eslint --cache --format=node_modules/eslint-formatter-pretty .",
-    "lint-fix": "prettier --single-quote --write \"{,!(flow-typed|dll)/**}/*.{js,jsx}\" && yarn lint -- --fix",
```
Comment (amilajack) [inline package.json]: `-- --someArg` is a way of propagating flags to npm script binaries. ex. `yarn lint -- --fix` will append the `--fix` flag to the `lint` script so that `eslint --cache --format=node_modules/eslint-formatter-pretty . --fix` will be run instead of `eslint --cache --format=node_modules/eslint-formatter-pretty .`. Was this change intentional?

Comment (jooohhn) [reply package.json]: Got this warning since I upgraded to yarn@v1.1.0: 


image

Removed the extra `--` and it went away. Thoughts?
